src/utils/daos_hdlr.c

```diff
@@ -1705,32 +1770,34 @@ cont_update_acl_hdlr(struct cmd_args_s *ap)
 	} else {
 		rc = daos_ace_from_str(ap->entry, &ace);
 		if (rc != 0) {
-			fprintf(stderr, "failed to parse entry: %d\n", rc);
+			fprintf(stderr, "failed to parse entry: %s (%d)\n",
+				d_errdesc(rc), rc);
 			return rc;
 		}
 
 		acl = daos_acl_create(&ace, 1);
 		daos_ace_free(ace);
 		if (acl == NULL) {
-			fprintf(stderr, "failed to make ACL from entry: %d\n",
-				rc);
+			fprintf(stderr, "failed to make ACL from entry: %s\n",
+				d_errdesc(rc));
```

Comment:
This one looks like it was a bug before your changes. If we got here, rc == 0.

Maybe we should update daos_acl_create to return an error code, because there are two reasons it might fail: invalid ACE list and out of memory. In this case we can assume OOM because the ACE was validated in daos_ace_from_str, and there is only one ACE. I think we should print the OOM error here for now.

Reply:
fixed